core/src/main/java/org/apache/jsieve/comparators/AsciiCasemap.java

```diff
@@ -40,6 +40,8 @@ public AsciiCasemap() {
      * @see org.apache.jsieve.comparators.Equals#equals(String, String)
      */
     public boolean equals(String string1, String string2) {
+	if (string1 == null && string2 == null) return true;
```

Comment:
I met this NPE in Zimbra version 8.7.1_GA_1670:
java.lang.NullPointerException
        at org.apache.jsieve.comparators.AsciiCasemap.equals(AsciiCasemap.java:43)
        at org.apache.jsieve.comparators.ComparatorUtils.is(ComparatorUtils.java:212)
        at org.apache.jsieve.comparators.ComparatorUtils.match(ComparatorUtils.java:63)
        at org.apache.jsieve.tests.Address.match(Address.java:88)
        at org.apache.jsieve.tests.Address.match(Address.java:52)
        at org.apache.jsieve.tests.AbstractComparatorTest.match(AbstractComparatorTest.java:197)
        at org.apache.jsieve.tests.AbstractComparatorTest.match(AbstractComparatorTest.java:169)
        at org.apache.jsieve.tests.AbstractComparatorTest.executeBasic(AbstractComparatorTest.java:144)
        at org.apache.jsieve.tests.AbstractTest.execute(AbstractTest.java:57)
        at org.apache.jsieve.Test.isTestPassed(Test.java:66)
        at org.apache.jsieve.TestList.anyTestsPass(TestList.java:112)
        at org.apache.jsieve.tests.AnyOf.executeBasic(AnyOf.java:45)
        at org.apache.jsieve.tests.AbstractTest.execute(AbstractTest.java:57)
        at org.apache.jsieve.Test.isTestPassed(Test.java:66)
        at org.apache.jsieve.TestList.allTestsPass(TestList.java:92)
        at org.apache.jsieve.commands.If.executeBasic(If.java:63)
        at org.apache.jsieve.commands.AbstractCommand.execute(AbstractCommand.java:122)
        at org.apache.jsieve.Command.execute(Command.java:158)
        at org.apache.jsieve.Commands.execute(Commands.java:83)
        at org.apache.jsieve.Block.execute(Block.java:80)
        at org.apache.jsieve.SieveParserVisitorImpl.visit(SieveParserVisitorImpl.java:218)
        at org.apache.jsieve.parser.generated.ASTstart.jjtAccept(ASTstart.java:18)
        at org.apache.jsieve.SieveFactory.evaluate(SieveFactory.java:155)
        at com.zimbra.cs.filter.RuleManager.applyRulesToExistingMessage(RuleManager.java:402)
        at com.zimbra.cs.service.mail.ApplyFilterRules.handle(ApplyFilterRules.java:150)
        at com.zimbra.soap.SoapEngine.dispatchRequest(SoapEngine.java:605)
        at com.zimbra.soap.SoapEngine.dispatch(SoapEngine.java:458)
        at com.zimbra.soap.SoapEngine.dispatch(SoapEngine.java:271)
        at com.zimbra.soap.SoapServlet.doWork(SoapServlet.java:303)
        at com.zimbra.soap.SoapServlet.doPost(SoapServlet.java:213)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
        at com.zimbra.cs.servlet.ZimbraServlet.service(ZimbraServlet.java:206)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:821)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1685)
        at com.zimbra.cs.servlet.CsrfFilter.doFilter(CsrfFilter.java:169)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
        at com.zimbra.cs.servlet.RequestStringFilter.doFilter(RequestStringFilter.java:54)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
        at com.zimbra.cs.servlet.SetHeaderFilter.doFilter(SetHeaderFilter.java:59)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
        at com.zimbra.cs.servlet.ETagHeaderFilter.doFilter(ETagHeaderFilter.java:47)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
        at com.zimbra.cs.servlet.ContextPathBasedThreadPoolBalancerFilter.doFilter(ContextPathBasedThreadPoolBalancerFilter.java:107)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
        at com.zimbra.cs.servlet.ZimbraQoSFilter.doFilter(ZimbraQoSFilter.java:116)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
        at com.zimbra.cs.servlet.ZimbraInvalidLoginFilter.doFilter(ZimbraInvalidLoginFilter.java:117)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
        at org.eclipse.jetty.servlets.DoSFilter.doFilterChain(DoSFilter.java:473)
        at org.eclipse.jetty.servlets.DoSFilter.doFilter(DoSFilter.java:318)
        at org.eclipse.jetty.servlets.DoSFilter.doFilter(DoSFilter.java:288)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:581)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1158)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1090)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:213)
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:109)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:119)
        at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:318)
        at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:437)
        at org.eclipse.jetty.server.handler.DebugHandler.handle(DebugHandler.java:84)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:119)
        at org.eclipse.jetty.server.Server.handle(Server.java:517)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:306)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:242)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:261)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)
        at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:192)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:261)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)
        at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:75)
        at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceAndRun(ExecuteProduceConsume.java:213)
        at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:147)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:654)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:572)
        at java.lang.Thread.run(Thread.java:745)